pytorch_transformers/tokenization_utils.py

```diff
@@ -696,15 +698,17 @@ def encode(self, text, text_pair=None, add_special_tokens=False, **kwargs):
         second_sentence_tokens = [self._convert_token_to_id(token) for token in self.tokenize(text_pair, **kwargs)]
 
         if add_special_tokens:
-            return self.add_special_tokens_sentences_pair(first_sentence_tokens, second_sentence_tokens)
+            return self.add_special_tokens_sentences_pair(first_sentence_tokens, second_sentence_tokens, output_mask)
         else:
+            if output_mask:
+                logger.warning("Can't output mask if no special tokens are involved. Please call the method with add_special_tokens set to True.")
```

Comment:
Do we really need `special_tokens` to have an output mask?

Reply:
Hmm this warning is put here so that if a user asks for an `output_mask` while not joining sequences (`add_special_tokens` is set to `False`), he is aware that no mask will be returned.

If the sequences are not joined together, I don't think it makes sense to output a mask,  does it?

Reply:
Maybe the warning can be something like `"Can't output mask if you're not joining two sequences"`